### DISCUSSION TITLE
What is the best way to provide user an option to specify all the args in file.

### DISCUSSION THREAD

**Discussion by semenoh** [ajalt/clikt#504] [Q&A]
Hi? I'm completely new to cli apps. This lib is quite extensive and feature reach so I assume this functionality is already there but it is not obvious to me. I'd like to make it possible to use my app it two ways: 
 - specify all the args in command line
 - specify a path to file where all the args are specified in some format.
So I'm wondering what is the best way to do so. And which format of the file to choose.

PS I'm thinking about redirecting input but I do not know neithe if it is good idea nor how to implement it.

**Comment by ajalt** [comment]
Clikt supports both [argument files][1] and custom [configuration files][2].

Where in the docs did you look for that info? Maybe we can update them to make that info easier to find.

[1]: https://ajalt.github.io/clikt/advanced/#command-line-argument-files-argfiles
[2]: https://ajalt.github.io/clikt/options/#values-from-configuration-files
